lighthouse-core/test/gather/driver/target-manager-test.js

```diff
@@ -118,22 +118,35 @@ describe('TargetManager', () => {
     });
 
     it('should listen to target before resuming', async () => {
+      let targetListeningAsserted = false;
+
+      // Intercept listener for all protocol events and ensure target is still paused.
       sessionMock.on = /** @type {typeof sessionMock.on} */ (fnAny()
         .mockImplementation(/** @param {string} eventName */ (eventName) => {
-          // Intercept listener for all protocol events and ensure target is still paused.
-          if (eventName === '*') {
-            expect(sendMock.findAllInvocations('Target.getTargetInfo')).toHaveLength(1);
-            expect(sendMock.findAllInvocations('Target.setAutoAttach')).toHaveLength(0);
-            expect(sendMock.findAllInvocations('Runtime.runIfWaitingForDebugger')).toHaveLength(0);
+          const getTargetInfoCount = sendMock.findAllInvocations('Target.getTargetInfo').length;
+          const setAutoAttachCount = sendMock.findAllInvocations('Target.setAutoAttach').length;
+          const resumeCount = sendMock.findAllInvocations('Runtime.runIfWaitingForDebugger').length;
+
+          // There may be many listeners for all protocol events, so just ensure this one occurred.
```

Comment:
This is now annoying, but still seems worth asserting. Open to other approaches.